core/src/rpc_pubsub.rs

```diff
@@ -921,13 +921,11 @@ mod tests {
         });
 
         // Process votes and check they were notified.
-        // FIX-ME-BETTER-LATER - clone below is required for testcase to pass
-        #[allow(clippy::redundant_clone)]
```

Comment:
@svenski123 Btw, do you know why not using `Arc<_>` solves the past instability of this test?

Reply:
It's not the Arc per se - it was taking ownership of the only handle to the subscription which was then dropped, closing the channel before anything was read.

The correct parameter passing convention here is that of borrowing, which ensure lifetime is at least that of the function and also avoids the need for the function to know anything about where the subscription is stored i.e. in an arc, a mutex, etc. - it just takes a reference.

Reply:
Oh, I see! That makes sense, then. :)